pkg/bpf/perf.go

```diff
@@ -295,21 +297,19 @@ func (e *PerfEvent) Disable() error {
 		return nil
 	}
 
+	C.free(e.state)
```

Comment:
Is `e.state.` guaranteed to never be `NULL`?

Reply:
No. We are using malloc so there is a chance it can return `NULL`. What is the proper way to handle this. Do you want to do a null check first?

Reply:
I guess we'd have to check on line 293 and return an error there. To be safe, we could make this no-op on `e.state` being nil too

Reply:
Will leave this as is unless @tgraf really want's it to change.

Reply:
I'm taking my comment back, calling free() on NULL is perfectly fine so we can keep this.